mbq/env/__init__.py

```diff
@@ -7,30 +9,73 @@ class EnvException(Exception):
     pass
 
 
+class Environment(Enum):
+    """
+    A simple Enum to provide consistency in how we detect and
+    represent the environment in which a service is running.
+    """
+
+    PRODUCTION = ('prd', 'Production')
+    DEVELOPMENT = ('dev', 'Development')
+    LOCAL = ('lcl', 'Local')
+
+    def __init__(self, short, long):
+        self.short = short
+        self.long = long
+        self._searchable = {short.lower(), long.lower()}
+
+    @classmethod
+    def from_value(cls, value):
+        """
+        Find an Environment using a single `str`.
+
+        Using a typical Enum, finding a member by value would be as
+        simple as MyEnum('the-value') but since Environment members
+        are stored as tuples and we want to look up using `str`s,
+        `from_value` can used to make that lookup in a case-
+        insensitive way.
+
+        >>> assert Environment.from_value('prd') is Environment.from_value('PRODUCTION')
+        """
+
+        for member in list(cls):
+            if value.lower() in member._searchable:
+                return member
+
+        raise ValueError(f"'{value}' is not a valid {cls.__name__}")
+
+
 class Env:
     def get(self, key, default=NOT_PROVIDED, required=True, coerce=NOT_PROVIDED):
         try:
             val = os.environ[key].strip()
-        except KeyError:
+        except KeyError as e:
             if default is not NOT_PROVIDED:
                 return default
 
             if not required:
                 return None
 
-            raise EnvException('Missing key "{}"'.format(key))
+            raise EnvException('Missing key "{}"'.format(key)) from e
```

Comment:
Whoa TIL

Reply:
Yeah it's a minor thing to make stack trace a bit easier to comprehend